lib/elixir/lib/keyword.ex

```diff
@@ -69,9 +69,7 @@ defmodule Keyword do
   """
   @spec new(Enum.t) :: t
   def new(pairs) do
-    Enum.reduce pairs, [], fn {k, v}, keywords ->
-      put(keywords, k, v)
-    end
+    Enum.uniq_by(Enum.reverse(pairs), fn {x, _} when is_atom(x) -> x end)
```

Comment:
We shouldn't expect keywords this big. So maybe a hand-written implementation would be better here.